src/Cmd.ml

```diff
@@ -55,39 +55,48 @@ let check t =
   match t with
   | Shell _ -> Promise.Result.return t
   | Spawn spawn ->
-    let open Promise.Result.O in
+    let open Promise.Result.Syntax in
     checkSpawn spawn >>| fun s -> Spawn s
 
 let run ?cwd ?stdin = function
   | Spawn { bin; args } ->
     ChildProcess.spawn (Path.toString bin) (Array.of_list args) ?stdin
-      (ChildProcess.Options.make ?cwd ())
+      (ChildProcess.Options.create ?cwd ())
   | Shell commandLine ->
-    ChildProcess.exec commandLine ?stdin (ChildProcess.Options.make ?cwd ())
+    ChildProcess.exec commandLine ?stdin (ChildProcess.Options.create ?cwd ())
 
 let log ?(result : ChildProcess.return option) (t : t) =
+  let open Jsonoo.Encode in
   let message =
     match result with
     | None -> []
-    | Some result -> [ ("result", Log.field result) ]
+    | Some result ->
+      [ ( "result"
+        , object_
+            [ ("exitCode", int result.exitCode)
+            ; ("stdout", string result.stdout)
+            ; ("stderr", string result.stderr)
+            ] )
+      ]
   in
   let message =
     match t with
     | Spawn { bin; args } ->
-      ("bin", Log.field (Path.toString bin))
-      :: ("args", Log.field (Json.Encode.list Json.Encode.string args))
+      ("bin", string (Path.toString bin))
+      :: ("args", list string args)
       :: message
-    | Shell commandLine -> ("shell", Log.field commandLine) :: message
+    | Shell commandLine -> ("shell", string commandLine) :: message
   in
   logJson "external command" message
 
 let output ?stdin (t : t) =
-  let open Promise.O in
+  let open Promise.Syntax in
```

Comment:
Why was this renamed?

Reply:
The `O` convention is what we use in dune to mean the same thing. I like it a little more than `Syntax` which is quite verbose. In any case, I don't think it matters here. I don't mind following core's convention.